src/main/java/com/github/zly2006/reden/rvc/tracking/TrackedStructure.kt

```diff
@@ -25,16 +34,197 @@ class TrackedStructure (
+                                IncludeEntry(
+                                    entry.points.filter { it.z < ignored.key.z }
+                                )
+                            )
+                            iter.add(
+                                IncludeEntry(
+                                    entry.points.filter { it.z > ignored.key.z }
+                                )
+                            )
+                        } else {
+                            var entryToSplit = entry
+                            iter.remove()
+                            // first, split by x
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.x < ignored.key.x }
+                                )
+                            )
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.x > ignored.key.x }
+                                )
+                            )
+                            // then add same x points to the new cuboids
+                            entryToSplit = IncludeEntry(entryToSplit.points.filter { it.x == ignored.key.x })
+                            if (entryToSplit.cuboid?.contains(ignored.key) != true) {
+                                iter.add(entryToSplit)
+                            }
+                            // second, split by y
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.y < ignored.key.y }
+                                )
+                            )
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.y > ignored.key.y }
+                                )
+                            )
+                            // then add same y points to the new cuboids
+                            entryToSplit = IncludeEntry(entryToSplit.points.filter { it.y == ignored.key.y })
+                            if (entryToSplit.cuboid?.contains(ignored.key) != true) {
+                                iter.add(entryToSplit)
+                            }
+                            // third, split by z
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.z < ignored.key.z }
+                                )
+                            )
+                            iter.add(
+                                IncludeEntry(
+                                    entryToSplit.points.filter { it.z > ignored.key.z }
+                                )
+                            )
+                            // then add same z point is ignored
+                        }
```

Comment:
The above implementation has many repeating or similar statements. Suggest extracting one or more methods from that.